### PR TITLE
Objectron to COCO fromat converter

### PR DESCRIPTION
**Pull request by sovrasov**

### CODE REVIEW

File: annotation_converters/objectron_2_coco.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+def load_video_info(data_root, subset, classes):
+    videos_info = []
+    avg_vid_len = 0
+    for cl in classes:
+        with open(osp.join(lists_root_path, f'{cl}_annotations_{subset}'), 'r') as f:
+            for line in f:
+                ann_path = osp.join(data_root, 'annotation' + osp.sep + line.strip() + '.pbdata')
+                ann = load_annotation_sequence(ann_path) # object class can be incorrect in annoatation
+                for item in ann:
+                    item[1] = cl
```
Comment (kprokofi) [inline annotation_converters/objectron_2_coco.py:24]: Have you ensured, that, in the case when class incorrect in annotation, everything else except the class name is correct?

File: annotation_converters/objectron_2_coco.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+        for frame_idx, frame_ann in enumerate(annotation):
+            if frame_idx not in frames:
+                continue
+            if frames[frame_idx] is None:
+                print('Warning: missing frame in ' + vid_path)
+                continue
+            #object_keypoints_2d, object_categories, keypoint_size_list, annotation_types
+
+            h, w = frames[frame_idx].shape[0] // res_divisor, frames[frame_idx].shape[1] // res_divisor
```
Comment (kprokofi) [inline annotation_converters/objectron_2_coco.py:109]: What if res_divisor is not 1? As I understand, then, the shape of the image would be changed and decoding of bbox coordinates would be with the new resolution, but they were encoded with the old one.

File: annotation_converters/objectron_2_coco.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    bboxes = []
+    num_valid = 0
+
+    for i in range(num_objects):
+        min_x = np.min(keypoints[i][:,0])
+        min_y = np.min(keypoints[i][:,1])
+        max_x = np.max(keypoints[i][:,0])
+        max_y = np.max(keypoints[i][:,1])
+        bbox = [min_x, min_y, max_x - min_x, max_y - min_y]
+        if min_x < 0 or min_y < 0 or max_x >= w or max_y >= h and bbox[2]*bbox[3] == 0:
```
Comment (kprokofi) [inline annotation_converters/objectron_2_coco.py:61]: maybe do not delete all of the bbox when they mismatch image resolution, but clamp coordinates first and then ensure that the length of the sides of the boxes does not shrink to zero.

Comment (sovrasov) [reply annotation_converters/objectron_2_coco.py:61]: If we do so without any threshold, a lot of hard outliers will arise. I do not observe critical for the detection task reduction of data, so prefer to keep the annotation clean.

File: annotation_converters/objectron_2_coco.py (hunk 2 above)
Comment (sovrasov) [reply annotation_converters/objectron_2_coco.py:109]: The formula of decoding in `decode_keypoints` is linear, so I guess I can decode with any resolution. Also I visually checked annotation with `res_divisor=2` and currently training a detector on it.

File: annotation_converters/objectron_2_coco.py (hunk 1 above)
Comment (sovrasov) [reply annotation_converters/objectron_2_coco.py:24]: Only for several videos. I'll add a warning message here to print names of suspectful videos. Later they could be manually checked.

File: annotation_converters/objectron_2_coco.py (hunk 3 above)
Comment (kprokofi) [reply annotation_converters/objectron_2_coco.py:61]: If there's no problem with data reduction - then okay

File: annotation_converters/objectron_2_coco.py (hunk 2 above)
Comment (kprokofi) [reply annotation_converters/objectron_2_coco.py:109]: Yes, seems so, I agree since it's linear

**Pull request merged**